app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb

```diff
@@ -0,0 +1,24 @@
+module ForemanTeamdynamix
+  module HostsHelperExtensions
+    extend ActiveSupport::Concern
+    DEFAULT_TD_PANE_FIELDS = { 'Asset ID': 'ID', 'Owner': 'OwningCustomerName', 'Parent Asset': 'ParentID' }
+
+    def td_tab_title
+      SETTINGS[:team_dynamix][:title] || 'Team Dynamix'
+    end
+
+    def teamdynamix_fields(host)
+      td_pane_fields = SETTINGS[:team_dynamix][:fields] || DEFAULT_TD_PANE_FIELDS
+      @asset = TeamDynamixApi.get_asset(host.asset_id)
```

Comment:
I think this field should be host.teamdynamix_asset_id so it is clear what plugin it comes from.
Also there will be many hosts without an associated assert at first so we should do something nicer if host.teamdynamix_asset_id is nil, like return this for fields or something [[_('Asset'), 'None Associated']]